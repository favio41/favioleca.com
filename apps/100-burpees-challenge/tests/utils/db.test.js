const DB = require("../../src/utils/db");

describe("DB", () => {
  test("keys", () => {
    expect(DB.keys).toMatchSnapshot();
  });

  test("should save", () => {
    const key = "test";
    DB.saveJson(key, { hello: "world" });
    expect(window.localStorage.getItem(key)).toMatchSnapshot();
  });

  test("should retrieve", () => {
    const key = "test";
    window.localStorage.setItem(key, JSON.stringify({ hello: "world" }));
    expect(DB.retrieveJson(key)).toMatchSnapshot();
  });
});
