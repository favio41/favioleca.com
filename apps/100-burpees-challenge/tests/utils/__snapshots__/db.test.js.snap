// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DB keys 1`] = `
{
  "LOGS": "BURPEES_LOGS",
  "USER_PREFERENCES": "BURPEES_USER_PREFERENCES",
}
`;

exports[`DB should retrieve 1`] = `
{
  "hello": "world",
}
`;

exports[`DB should save 1`] = `"{"hello":"world"}"`;
